src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureFlagOptions.cs

```diff
@@ -27,7 +38,26 @@ public class FeatureFlagOptions
         /// <summary>
         /// The time after which the cached values of the feature flags expire.  Must be greater than or equal to 1 second.
         /// </summary>
-        public TimeSpan CacheExpirationInterval { get; set; } = RefreshConstants.DefaultFeatureFlagsCacheExpirationInterval;
+        [Obsolete("The CacheExpirationInterval property will be deprecated in a future release. Please use the new `SetFeatureFlagRefreshInterval` method instead. " +
+            "Note that the usage has changed, but the functionality remains the same.")]
+        public TimeSpan CacheExpirationInterval
+        {
+            get { return _refreshInterval; }
+            set { _refreshInterval = value; }
+        }
+
+        /// <summary>
+        /// Sets the time after which feature flags can be refreshed.
+        /// </summary>
+        /// <param name="refreshInterval">
+        /// The minimum time after which feature flags can be refreshed. Must be greater than or equal to 1 second.
+        /// </param>
+        public FeatureFlagOptions SetFeatureFlagRefreshInterval(TimeSpan refreshInterval)
```

Comment:
AFAIU here besides the renaming, it also removes the public getter. Is it by design?

Reply:
Just wondering are there any existing test cases setting/getting `CacheExpirationInterval`? And do we want to also add cases for this new API?

Reply:
I think we can just use `SetRefreshInterval`. It's under the `FeatureFlagOptions`, so it shouldn't cause any confusion.

Reply:
> Just wondering are there any existing test cases setting/getting `CacheExpirationInterval`? And do we want to also add cases for this new API?

Good point actually, I didn't re-update the tests when I made some changes, but it would make sense to have at least some reference the old getter and make sure the rest use the API instead of the internal property.